### PR TITLE
feat : add code for ChildEventListener

### LINKED ISSUE #7: Code snippet for ChildEventListener
Adding code snippet for ChildEventListener in Java and Kotlin.

@samtstern working on this, will send a PR :)

### PR DESCRIPTION
**Pull request by the-dagger**
Fixes #7 
@samtstern Modified the existing code for valueEventListener as well.
Please take a look and let me know if that looks good.

Best!

### CODE REVIEW

File: database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java [hunk 1]
```diff
@@ -29,32 +30,35 @@
...
         setContentView(R.layout.activity_main);
+        FirebaseDatabase database = FirebaseDatabase.getInstance();
+        myRef = database.getReference("message");
     }
 
     @Override
-    public void basicReadWrite() {
-        // [START write_message]
-        // Write a message to the database
-        FirebaseDatabase database = FirebaseDatabase.getInstance();
```
Comment (samtstern) [inline database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java]: We actually do want to show the initialization of the `database` and `myRef` variables as part of the snippet here.  For many people this code snippet is the first Firebase code they ever see and we want to be very explicit.

File: database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java [hunk 2]
```diff
@@ -29,32 +30,35 @@
...
-        DatabaseReference myRef = database.getReference("message");
-
-        myRef.setValue("Hello, World!");
-        // [END write_message]
-
+    public void valueRead() {
         // [START read_message]
         // Read from the database
+        /*
```
Comment (samtstern) [inline database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java]: I appreciate the improvement you're trying to make, but I think this comment is too much.  Could you remove it?  We should be covering this material in the documentation that surrounds the snippets.

File: database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java [hunk 3]
```diff
@@ -29,32 +30,35 @@
...
+        so every time the data on myRef changes, you get a snapshot of the entire myRef node.
+         */
         myRef.addValueEventListener(new ValueEventListener() {
             @Override
             public void onDataChange(DataSnapshot dataSnapshot) {
                 // This method is called once with the initial value and again
                 // whenever data at this location is updated.
-                String value = dataSnapshot.getValue(String.class);
-                Log.d(TAG, "Value is: " + value);
+                // Since we get a snapshot of the data we create an Iterable to traverse through it.
```
Comment (samtstern) [inline database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java]: Why did you make this change? We wrote "Hello World" to the reference so it's a String not an Iterable

File: database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java [hunk 4]
```diff
@@ -65,4 +69,63 @@ public void onCancelled(DatabaseError error) {
         });
         // [END read_message]
     }
+
+    @Override
+    public void childRead() {
+        // [START read_message]
+        // Read from the database
+
+        /*
```
Comment (samtstern) [inline database/app/src/main/java/com/google/firebase/referencecode/database/MainActivity.java]: Similar to above, let's remove this larger comment.

File: database/app/src/main/java/com/google/firebase/referencecode/database/KotlinMainActivity.kt [hunk 5]
```diff
@@ -48,4 +48,54 @@ class KotlinMainActivity : AppCompatActivity(), MainActivityInterface {
         // [END read_message]
     }
 
+    override fun childRead() {
+        // [START read_message]
```
Comment (samtstern) [inline database/app/src/main/java/com/google/firebase/referencecode/database/KotlinMainActivity.kt]: The tag name after `START` must be uniqe in the file

**Comment by samtstern** [comment]
@the-dagger thanks for sending!  I made some inline comments but I am a little confused.  Where did this child event listener snippet come from?  Is it represented in the docs somewhere?

The snippets here are only for the purpose of being embedded in our documentation.  I apologize if I misled you!

**Comment by the-dagger** [comment]
Hi @samtstern Looks like a mistake on my end.
I didn't realize that this was meant to be a reference for docs.

I'll stick to converting the existing snippets to kotlin and you can go ahead and add new snippets!

**Comment by samtstern** [comment]
@the-dagger sorry that was a miscommunication on my part, thanks for understanding!

If you're curious, here's basically how it works:

  * Our docs include template tags like `{% include github_path="firebase/snippets-android/blah/blah/File.java" region_tag="basic_read" %}`
  * The docs builder goes to Github, extracts the part of the code between `[START basic_read]` and `[END basic_read]`.
  * The snippets appear on the page, like this: https://firebase.google.com/docs/database/android/start/

So these `snippets-*` repos are just places we can put these snippets so that we can test them, compile them, etc.  Keeps our docs healthier than having code right in the HTML!

If you want to add new snippets, it should start with the place in the docs where you think they're useful.  But if you do see opportunities like this, your feedback is always welcome and we can work on making improvements.

**Comment by the-dagger** [comment]
@samtstern Ah ok, I got it thanks! 👍 
What I can do for the time being is convert the existing Java snippets to Kotlin.
Also I think that it's a good idea that you go ahead and add the code snippets and I will convert them to Kotlin.

How does that sound?

**Comment by samtstern** [comment]
@the-dagger that sounds good, I think a lot of the snippets on the site are currently pulling from the quickstart-android repo but maybe we could move some of them here so that we can also have them in Kotlin, since I don't (yet) want to do Kotlin quickstarts.

**Comment by the-dagger** [comment]
That's great!
I would definitely love to help with converting them to kotlin.
In the meantime, can I convert the existing snippets here to Kotlin like I did in #6  ?

**Comment by samtstern** [comment]
@the-dagger yes any snippet you see in this repo and want to convert to Kotlin, go for it!  I am happy to review them.

**Pull request closed without merging**